source/E2E.Tests/Services/PartyBases/PartyBaseSyncTests.cs

```diff
@@ -0,0 +1,268 @@
+﻿using Autofac.Features.OwnedInstances;
+using E2E.Tests.Environment;
+using E2E.Tests.Environment.Instance;
+using E2E.Tests.Util;
+using HarmonyLib;
+using System.Runtime.InteropServices;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.CampaignSystem.MapEvents;
+using TaleWorlds.CampaignSystem.Party;
+using TaleWorlds.CampaignSystem.Roster;
+using TaleWorlds.CampaignSystem.Settlements;
+using TaleWorlds.CampaignSystem.Settlements.Buildings;
+using Xunit.Abstractions;
+
+namespace E2E.Tests.Services.PartyBases;
+
+public class PartyBaseSyncTests : IDisposable
+{
+    E2ETestEnvironment TestEnvironment { get; }
+
+    EnvironmentInstance Server => TestEnvironment.Server;
+
+    IEnumerable<EnvironmentInstance> Clients => TestEnvironment.Clients;
+
+    private string PartyId;
+    private string ItemRosterId;
+    private string MapEventSideId;
+    private string SettlementId;
+    private string MemberRosterId;
+    private string PrisonRosterId;
+    private string HeroId;
+    private string PartyBaseId;
+
+    MobileParty party;
+    ItemRoster itemRoster;
+    MapEventSide mapEventSide;
+    Settlement settlement;
+    TroopRoster memberRoster;
+    TroopRoster prisonRoster;
+    Hero hero;
+    PartyBase partyBase;
+
+    public PartyBaseSyncTests(ITestOutputHelper output)
+    {
+        TestEnvironment = new E2ETestEnvironment(output);
+    }
+
+    public void Dispose()
+    {
+        TestEnvironment.Dispose();
+    }
+
+    [Fact]
+    public void Server_MobileParty_SyncAllClients()
+    {
+        // Arrange
+        var server = TestEnvironment.Server;
+
+        // Act
+        var customOwner = AccessTools.Field(typeof(PartyBase), nameof(PartyBase._customOwner));
+        var index = AccessTools.Field(typeof(PartyBase), nameof(PartyBase._index));
+        var lastEatingTime = AccessTools.Field(typeof(PartyBase), nameof(PartyBase._lastEatingTime));
+        var lastMenPerTierVerNo = AccessTools.Field(typeof(PartyBase), nameof(PartyBase._lastNumberOfMenPerTierVersionNo));
+        var mapEventSideField = AccessTools.Field(typeof(PartyBase), nameof(PartyBase._mapEventSide));
+        var numberMenHorseField = AccessTools.Field(typeof(PartyBase), nameof(PartyBase._numberOfMenWithHorse));
+        var remainingFoodPercentageField = AccessTools.Field(typeof(PartyBase), nameof(PartyBase._remainingFoodPercentage));
+        var lastNumberMenHorseVerionField = AccessTools.Field(typeof(PartyBase), nameof(PartyBase._lastNumberOfMenWithHorseVersionNo));
```

Comment:
Missing _prisonerSizeLastCheckVersion

Reply:
I removed this on purpose, if we sync this we also have to sync all the cached variables whereas we would instead just let the client calculate that based on the values that the server has previously provided.

Reply:
Can you update the issue to reflect this decision?